api/integrations/slack/pagination.py

```diff
@@ -0,0 +1,22 @@
+from collections import OrderedDict
+
+from rest_framework.pagination import CursorPagination
+from rest_framework.response import Response
+
+
+class ChannelListPagination(CursorPagination):
+    page_size_query_param = "page_size"
+
+    def paginate_queryset(self, queryset, request, view=None):
+        page_size = self.get_page_size(request)
+        cursor = request.GET.get("cursor", None)
+        response = queryset.get_channels_data(limit=page_size, cursor=cursor)
```

Comment:
I do think this use of the pagination class is quite neat but I'm not sure about our use of `queryset` here. It's a bit misleading to have a queryset that is not iterable. I think we should rethink this slightly. Maybe `get_channels_data` should return some kind of iterable?

Reply:
The more I think about this, I don't think we're approaching it correctly. We're not _Really_ doing the pagination. That's why it feels a little bit like we're shoehorning our logic into a django pagination class. 

Essentially all we're doing is proxying a request and doing some serialization. 

I think this code (views, serializers, pagination) would be neater if we just dropped all of the pagination stuff and just had an endpoint that accepts some get parameters and returns some serialized data. 

Something along these lines?

```
# serializers.py

class GetChannelsQuerySerializer(serializers.Serializer):
    cursor = serializers.CharField()
    limit = serializers.IntegerField()

# views.py

class SlackChannelsViewSet(GenericViewSet):
    def list(self, request, *args, **kwargs):
        query_serializer = GetChannelsQuerySerializer(data=request.data)
        query_serializer.is_valid(raise_exception=True)
        
        api_token = self.get_api_token()
        slack_wrapper = SlackWrapper(api_token=api_token)
        
        channels_data = slack_wrapper.get_channels_data(**query_serializer.data)
        return self.get_serializer(channels_data)
```

Reply:
yeah, that makes sense